nilearn/decomposition/base_msdl.py

```diff
@@ -0,0 +1,511 @@
+                        tol=1e-4,
+                        max_iter=300,
+                        n_shuffle_subjects=None,
+                        n_jobs=1,
+                        verbose=False):
+        self.n_components = n_components
+        self.prox = prox
+        self.alpha = alpha
+        self.mu = mu
+        self.mem = mem
+        self.max_iter = max_iter
+        self.verbose = verbose
+        self.non_penalized = non_penalized
+        self.do_ica = do_ica
+        self.tol = tol
+        self.n_shuffle_subjects = n_shuffle_subjects
+        self.n_jobs = n_jobs
+        self.mask = mask
+
+    def fit(self, data, V_init=None, **params):
+        self.mask_ = self.mask.get_data().astype(np.bool)
+        mu = self.mu
+        if V_init is None:
+            imgs = [unmask(d, self.mask) for d in data]
+            if self.do_ica:
+                init_model = CanICA(mask=self.mask,
+                                n_components=self.n_components,
+                                memory=self.mem,
+                                n_jobs=self.n_jobs)
+            else:
+                raise NotImplementedError
+                # We should use a multi-pca instead of the ICA
+            init_model.fit(imgs)
+            V_init = init_model.components_
+
+            # Store for later inspection
+            self.init_model_ = init_model
+            del imgs
+            # We may be imposing a non-negativity constraint, thus we
+            # need to make sure that the initial components are as
+            # positive as possible
+            for component in V_init:
+                # Note that the components are centered, thus if we
+                # do not square, the two sums will be equal
+                if (np.sum(component[component > 0] ** 2) <
+                        np.sum(component[component < 0] ** 2)):
+                    component[:] *= -1
+            if self.verbose > 1:
+                print('[%s] Computed V_init' % self.__class__.__name__)
+        Us = list()
+        Ss = list()
+
+        if mu == 'auto':
+            mu = self.mem.cache(estimate_group_regularization,
+                                ignore=['mem'])(data,
+                                self.n_components, self.mem)
+            self.mu_ = mu
+
+        for Y_ in data:
+            # Relearn U, V to have the right scaling on U
+            Y__ = Y_
+            U = linalg.lstsq(V_init.T, Y__.T)[0].copy()
+            del Y__
+            S = np.sqrt((U ** 2).sum(axis=1))
+            S[S == 0] = 1
+            U /= S[:, np.newaxis]
+            Ss.append(S)
+            Us.append(U.T)
+        V_init = (V_init * np.mean(Ss, axis=0)[:, np.newaxis])
+        if self.verbose > 1:
+            print('[%s] Finished learning U_init' % self.__class__.__name__)
+        if self.verbose > 1:
+            print('[%s] Finished learning V_init' % self.__class__.__name__)
+
+        _, maps_, group_maps_, E = group_dictionary(data,
```

Comment:
If kept, it has to be renamed. The question is: Should we keep it? Nowadays I tend to prefer outputting that kind of information in stdout (in verbose mode) and using grep afterward.